terraform/modules/user_groups/main.tf

```diff
@@ -0,0 +1,75 @@
+# user_groups/main.tf
+
+variable "region" {
+  description = "AWS region where the user group will be created"
+}
+
+resource "aws_iam_group" "developer_group" {
+  name = "${var.region}-developer"
+}
+
+resource "aws_iam_policy" "full_access_policy" {
+  name        = "${var.region}-full-access-policy"
+  description = "Policy granting full access to AWS services in ${var.region}"
+  policy      = <<EOF
+{
+  "Version": "2012-10-17",
+  "Statement": [
+    {
+      "Effect": "Allow",
+      "Action": "*",
+      "Resource": "*"
+    }
+  ]
+}
+EOF
+}
+
+resource "aws_iam_policy_attachment" "attach_full_access_policy" {
+  name       = "${var.region}-full-access-attachment"
+  policy_arn = aws_iam_policy.full_access_policy.arn
+  groups     = [aws_iam_group.developer_group.name]
+}
+
+# Explicit permissions for each service
+resource "aws_iam_policy_attachment" "ecs_policy_attachment" {
+  name       = "${var.region}-ecs-attachment"
+  policy_arn = "arn:aws:iam::aws:policy/AmazonECS_FullAccess"
```

Comment:
Would it be possible to restrict to an area? You might have to write a custom policy for this.

Reply:
Same with the ones below

Reply:
> Would it be possible to restrict to an area? You might have to write a custom policy for this.

you mean instead of writing policy attachments, just writing our own custom policy?

Reply:
> > Would it be possible to restrict to an area? You might have to write a custom policy for this.
> 
> you mean instead of writing policy attachments, just writing our own custom policy? Like i did for full_access_policy on line 5?

Like ive done in my latest commit you mean?

Reply:
yea but the resource should be restricted to that region, if possible

Reply:
> yea but the resource should be restricted to that region, if possible

isnt it happening even now? I might be wrong but since ive passed the region as a variable, and all the policies are for that region, doesnt that mean that theyre already restricted to that?